src/com.docker.slirp/src/main.ml

```diff
@@ -120,86 +184,131 @@ let main_t socket_path port_control_path max_connections vsock_path db_path nofi
 end
 
-let main socket port_control max_connections vsock_path db nofile pcap select debug =
+let main socket port_control max_connections vsock_path db_path dns pcap select debug =
   let module Use_lwt_unix = Main(Host_lwt_unix) in
   let module Use_uwt = Main(Host_uwt) in
   (if select then Use_lwt_unix.main else Use_uwt.main)
-    socket port_control max_connections vsock_path db nofile pcap debug
+    socket port_control max_connections vsock_path db_path dns pcap debug
 
 open Cmdliner
 
-(* NOTE(aduermael): it seems to me that "/var/tmp/com.docker.slirp.socket" is a default value, right?
-This socket path is now dynamic, depending on current user's home directory. Could we just
-make it fail instead? In case no argument is supplied? *)
 let socket =
-  Arg.(value & opt string "/var/tmp/com.docker.slirp.socket" & info [ "socket" ] ~docv:"SOCKET")
+  let doc =
+    Arg.info ~doc:
+      "A URLs to connect to for ethernet of the form \
+      hyperv-connect://vmid/serviceid or hyperv-connect://vmid for the default serviceid on Windows \
+      or /var/tmp/com.docker.slirp.socket on Mac" ["ethernet"]
+  in
+  Arg.(value & opt string "" doc)
 
-(* NOTE(aduermael): it seems to me that "/var/tmp/com.docker.port.socket" is a default value, right?
-This socket path is now dynamic, depending on current user's home directory. Could we just
-make it fail instead? In case no argument is supplied? *)
 let port_control_path =
-  Arg.(value & opt string "/var/tmp/com.docker.port.socket" & info [ "port-control" ] ~docv:"PORT")
+  let doc =
+    Arg.info ~doc:
+      "A URL to connect to for port control of the form \
+     hyperv-connect://vmid/serviceid on Windows or \
+     /var/tmp/com.docker.port.socket on Mac" ["port"]
+  in
+  Arg.(value & opt string "" doc)
 
 let max_connections =
- let doc =
-   Arg.info ~doc:
-     "Maximum number of concurrent forwarded connections" [ "max-connections" ]
- in
- Arg.(value & opt (some int) None doc)
-
-(* NOTE(aduermael): it seems to me that "/var/tmp/com.docker.vsock/connect" is a default value, right?
-This socket path is now dynamic, depending on current user's home directory. Could we just
-make it fail instead? In case no argument is supplied? *)
+  let doc =
+    Arg.info ~doc:
+      "Maximum number of concurrent forwarded connections" [ "max-connections" ]
+  in
+  Arg.(value & opt (some int) None doc)
+
 let vsock_path =
-  Arg.(value & opt string "/var/tmp/com.docker.vsock/connect" & info [ "vsock-path" ] ~docv:"VSOCK")
+  let doc =
+    Arg.info ~doc:
+      "Path of the Unix domain socket used to setup virtio-vsock connections \
+       to the VM." [ "vsock-path" ] ~docv:"VSOCK"
+  in
+  Arg.(value & opt string "" doc)
 
 let db_path =
-  Arg.(value & opt (some string) None & info [ "db" ] ~docv:"DB")
+  let doc =
+    Arg.info ~doc:
+      "A URLs to connect to datakitof the form \
+      file:///var/tmp/foo or tcp://host:port or \\\\\\\\.\\\\pipe\\\\irmin" ["db"]
```

Comment:
URLs -> URL, I think, and "datakitof" -> "datakit of", and could also use rephrasing as above.